rcsbapi/search/search_query.py

```diff
@@ -1094,6 +1098,142 @@ def __str__(self):
             raise ValueError("Illegal Operator")
 
 
+class NestedAttributeQuery(Group):
+    """
+    Represents a search query restricted to a single nested attribute group.
+
+    Ensures that all attributes in the query fall within a valid nested indexing context, and provides
+    consistent behavior for nesting logic. It wraps `AttributeQuery` objects and restricts operations to those
+    allowed by the nested schema rules.
+
+    Args:
+        AttributeQuery (AttributeQuery): Two attribute-level queries that must all belong to the same nested context group.
+
+    Raises:
+        Warning: If queries do not all belong to a valid and consistent nested attribute group.
```

Comment:
Does it raise a warning or an error? Can you please confirm if `logger.error` actually raises an error and interrupts the process, or if it just logs and error and continues? If the latter, we should raise an error.